flake.nix

```diff
@@ -33,7 +34,31 @@
           cinaps = "*";
           ocamlfind = "1.9.2";
         };
-        pkgs = nixpkgs.legacyPackages.${system};
+        pkgs = nixpkgs.legacyPackages.${system}.extend (self: super: {
+          ocaml-ng = super.ocaml-ng // {
+            ocamlPackages_4_14 =
+              super.ocaml-ng.ocamlPackages_4_14.overrideScope' (oself: osuper: {
+                dune_3 = osuper.dune_3.overrideAttrs (o: {
```

Comment:
should only be needed until https://github.com/NixOS/nixpkgs/pull/196899 is merged.